core/src/main/java/jenkins/model/item_category/ItemCategory.java

```diff
@@ -0,0 +1,116 @@
+package jenkins.model.item_category;
+
+import hudson.Extension;
+import hudson.ExtensionList;
+import hudson.ExtensionPoint;
+import hudson.model.ModelObject;
+import hudson.model.TopLevelItemDescriptor;
+import jenkins.model.Messages;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import java.util.Collection;
+
+/**
+ * A category for {@link hudson.model.Item}s.
+ *
+ * @since TODO
+ */
+public abstract class ItemCategory implements ExtensionPoint {
+
+    public static int MIN_TOSHOW = 1;
+
+    /**
+     * Helpful to set the order.
+     */
+    private int weight = 1;
+
+    /**
+     * Identifier, e.g. "standaloneprojects", etc.
+     *
+     * @return the identifier
+     */
+    public abstract String getId();
+
+    /**
+     * The description in plain text
+     *
+     * @return the description
+     */
+    public abstract String getDescription();
+
+    /**
+     * A human readable name.
+     *
+     * @return the display name
+     */
+    public abstract String getDisplayName();
+
+    /**
+     * Minimum number required to show the category.
+     *
+     * @return the minimum items required
+     */
+    public abstract int getMinToShow();
+
+    protected void setWeight(int weight) {
```

Comment:
@jglick I needed to store the value here:
```
    public static ItemCategory getCategory(TopLevelItemDescriptor descriptor) {
        int weight = 1;
        ExtensionList<ItemCategory> categories = ExtensionList.lookup(ItemCategory.class);
        for (ItemCategory category : categories) {
            if (category.getId().equals(descriptor.getCategoryId())) {
                category.setWeight(categories.size() - weight);
                return category;
            }
            weight++;
        }
        return new UncategorizedCategory();
    }
```

I decided to do it `protected` thinking on different implementation of `getCategory()`. Imagine you want to override this method and you need to use `setWeight()`...